lit_nlp/examples/models/tydi.py

```diff
@@ -0,0 +1,97 @@
+"""LIT wrappers for TyDiModel"""
+from lit_nlp.api import model as lit_model
+from lit_nlp.api import types as lit_types
+import transformers
+
+
+
+BertTokenizer = transformers.BertTokenizer
+FlaxBertForQuestionAnswering = transformers.FlaxBertForQuestionAnswering
+JsonDict = lit_types.JsonDict
+
+
+def validate_TyDiModel(model: lit_model.Model) -> lit_model.Model:
+  """Validate that a given model looks like a TyDi model used by tydi_test.py.
+  Args:
+    model: a LIT model
+
+  Returns:
+    model: the same model
+
+  Raises:
+    AssertionError: if the model's spec does not match that expected for a TyDi
+    model.
+  """
+  # Check inputs
+  ispec = model.input_spec()
+  assert "context" in ispec
+  assert isinstance(ispec["context"], lit_types.TextSegment)
+  if "answers_text" in ispec:
+    assert isinstance(ispec["answers_text"], lit_types.MultiSegmentAnnotations)
+
+  # Check outputs
+  ospec = model.output_spec()
+  assert "generated_text" in ospec
+  assert isinstance(
+      ospec["generated_text"],
+      (lit_types.GeneratedText))
+  assert ospec["generated_text"].parent == "answers_text"
+
+  return model
```

Comment:
Move this logic to tydi_test.py. 

The logic itself looks good enough for this demo, but since we don't use this function anywhere else inside this file, the logic probably belongs inside the test for this class.

T5 is an exception to this because the `SummarizationWrapper` and `TranslationWrapper` classes take an already initialized model as a parameter, and have to validate that model against a spec to ensure compatibility before doing the rest of their initialization. Since `TyDiModel` doesn't have any wrapper classes, we don't need to validate internally.